src/main/perl/Fetch.pm

```diff
@@ -183,6 +183,7 @@ sub fetchProfile
     %current = $self->current($profile, %previous);
     if ($self->process_profile("$profile", %current) == $ERROR) {
         $self->error("Failed to process profile for $self->{PROFILE_URL}");
+        $current{cid}->cancel();
```

Comment:
The previous has its content set after this return, so nothing happens. But I've put it into a _cleanup method and called that from both places.